libs/shared/src/types/rate-limiting/index.ts

```diff
@@ -7,3 +7,26 @@ type ApiRateLimitNamespace = 'API_RATE_LIMIT';
 
 export type ApiRateLimitEnvVarFormat =
   Uppercase<`${ApiRateLimitNamespace}_${ApiServiceLevelTypeEnum}_${ApiRateLimitCategoryTypeEnum}`>;
+
+export enum ApiRateLimitConfigurationEnum {
+  BURST_ALLOWANCE = 'burstAllowance',
+  REFILL_INTERVAL = 'refillInterval',
+}
+
+export class IApiRateLimitConfiguration implements Record<ApiRateLimitConfigurationEnum, unknown> {
```

Comment:
This `implements` pattern allows us to create a typing that strictly required all class attributes have keys in the `ApiRateLimitConfigurationEnum`.